tensorflow/python/ops/special_math_ops.py

```diff
@@ -105,37 +107,43 @@ def einsum(axes, *inputs):
   idx_in = match.group(1).split(',')
   idx_out = match.group(2)
   idx_all = set(''.join(idx_in))
-
+  indices = str(idx_all)
 
   assert len(idx_in) == len(inputs), \
     "Expected %d inputs but only got %d" % (len(idx_in), len(inputs))
 
-  # transpose inputs so axes are in alphabetical order
+  missing_idx = set(idx_out).difference(idx_all)
+  assert not missing_idx, \
+    "Unknown ouput axes: %s" % missing_idx
```

Comment:
Can you make this into an exception (ValueError is fine), and add it to the docstring?